control/tests/test_heal.py

```diff
@@ -1,36 +1,150 @@
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+            'inst_class': {}
+        }
+
+    )
+    @unpack
+    def test_determine_fault(self, edx_id, video_trans_status, video_active, inst_class, video_trans_start):
+        """
+        Tests that determine_fault works as expected.
+        """
+        video_instance = Video(
+            edx_id=edx_id,
+            video_trans_status=video_trans_status,
+            video_trans_start=video_trans_start,
+            video_active=video_active,
+            inst_class=Course()
+        )
+        encode_list = self.heal_instance.determine_fault(video_instance)
 
-    def test_encode_url(self):
-        H = VedaHeal()
-        H.discovery()
+        if video_instance.edx_id == '1':
+            self.assertEqual(encode_list, [])
+        elif video_instance.edx_id == '2':
+            for e in encode_list:
+                self.assertTrue(e in self.encode_list)
 
+    @data(
+        {
+            'uncompleted_encodes': [],
+            'expected_encodes': ['test_obj'],
+            'video_object': {
+                'edx_id': '1',
+                'video_trans_status': 'Complete',
+                'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+                'video_active': True,
+                'inst_class': {
+                    'review_proc': True
+                }
+            }
+        },
+        {
+            'uncompleted_encodes': ['test_obj'],
+            'expected_encodes': ['test_obj'],
+            'video_object': {
+                'edx_id': '2',
+                'video_trans_status': 'Ingest',
+                'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+                'video_active': True,
+                'inst_class': {
+                    'review_proc': True
+                }
+            }
+        }
 
-def main():
-    unittest.main()
+    )
+    @unpack
+    def test_differentiate_encodes(self, uncompleted_encodes, expected_encodes, video_object):
+        """
+        Tests that differentiate_encodes works as expected.
+        """
+        video_instance = Video(
+            edx_id=video_object['edx_id'],
+            video_trans_status=video_object['video_trans_status'],
+            video_trans_start=video_object['video_trans_start'],
+            video_active=video_object['video_active'],
+            inst_class=Course()
+        )
+        video_instance.inst_class.review_proc = video_object['inst_class']['review_proc']
 
+        encode_list = self.heal_instance.differentiate_encodes(
+            uncompleted_encodes,
+            expected_encodes,
+            video_instance
+        )
 
-if __name__ == '__main__':
-    sys.exit(main())
+        if video_instance.edx_id == '1':
+            self.assertEqual(encode_list, [])
+        elif video_instance.edx_id == '2':
+            self.assertEqual(encode_list, ['test_obj'])
```

Comment:
Can we add test about determine_longterm_corrupt as well. Most of the code would be same but it would be nice if we have some for different cases.

Reply:
👍